forecasting/src/autogluon/forecasting/models/gluonts_model/abstract_gluonts/abstract_gluonts_model.py

```diff
@@ -0,0 +1,136 @@
+
+    def __init__(self, path: str, freq: str, prediction_length: int, name: str, eval_metric: str = None,
+                 hyperparameters=None, model=None):
+        """
+        Create a new model
+        Args:
+            path(str): directory where to store all the model
+            name(str): name of subdirectory inside path where model will be saved.
+            eval_metric(str): objective function the model intends to optimize, will use mean_wQuantileLoss by default
+            hyperparameters: various hyperparameters that will be used by model (can be search spaces instead of fixed values).
+        """
+        super().__init__()
+        self.name = name
+        self.path_root = path
+        self.path_suffix = self.name + os.path.sep
+        self.path = self.path_root + self.path_suffix
+
+        if eval_metric is None:
+            eval_metric = "mean_wQuantileLoss"
+        if eval_metric is not None and eval_metric not in ["MASE", "MAPE", "sMAPE", "mean_wQuantileLoss"]:
+            raise ValueError(f"metric {eval_metric} is not available yet.")
+        self.eval_metric = eval_metric
+
+        self.params = {}
+        self.set_default_parameters()
+        self.params["freq"] = freq
+        self.params["prediction_length"] = prediction_length
+        self.nondefault_parameters = []
+        if hyperparameters is not None:
+            self.params.update(hyperparameters)
+            self.nondefault_parameters = list(hyperparameters.keys())[:]
+
+        self.model = model
+        self.val_score = None
+        self.fit_time = None
+        self.predict_time = None
+
+    def set_contexts(self, path_context):
+        self.path = self.create_contexts(path_context)
+        self.path_suffix = self.name + os.path.sep
+        self.path_root = self.path.rsplit(self.path_suffix, 1)[0]
+
+    @staticmethod
+    def create_contexts(path_context):
+        path = path_context
+        return path
+
+    def save(self, path: str = None,):
+        if path is None:
+            path = self.path
+        file_path = path + self.model_file_name
+        save_pkl.save(path=file_path, object=self)
+        return path
+
+    @classmethod
+    def load(cls, path: str, reset_path=True):
+        file_path = path + cls.model_file_name
+        model = load_pkl.load(path=file_path,)
+        if reset_path:
+            model.set_context(path)
+        return model
+
+    def set_default_parameters(self):
+        pass
+
+    def create_model(self):
+        pass
+
+    def fit(self, train_data, time_limit=None):
+        pass
+
+    def predict(self, data, quantiles=None):
+        if quantiles is None:
+            quantiles = [0.5]
+        result_dict = {}
+        predicted_targets = list(self.model.predict(data))
+        status = [0 < quantiles[i] < 1 and str(quantiles[i]) in predicted_targets[0].forecast_keys for i in range(len(quantiles))]
+        if not all(status):
+            raise ValueError("Invalid quantile value.")
```

Comment:
So the logic for input quantiles is the following: There are two places where the users can input quantiles: when the users call task.fit(), and when users call predictor.predict(). The quantile in task.fit() decides what the quantiles that each model will actually produce, and the quantile in predictor.predict() decides what the quantiles will be shown to the users. If the quantile in task.fit() is None, each model will by default predict the quantile for [0.5]; if the quantile in predictor is None, then it will by default present all available quantiles(will present [0.5] only when there is no input for quantile in task.fit())